apps/meteor/ee/app/livechat-enterprise/server/api/rooms.ts

```diff
@@ -1,57 +1,108 @@
-			if (!room || room.t !== 'l') {
-				return API.v1.failure('Invalid room Id');
+			const room = await LivechatRooms.findOneById<Pick<IOmnichannelRoom, '_id' | 't' | 'open' | 'onHold' | 'lastMessage' | 'servedBy'>>(
+				roomId,
+				{
+					projection: { _id: 1, t: 1, open: 1, onHold: 1, lastMessage: 1, servedBy: 1 },
+				},
+			);
+			if (!room || !isOmnichannelRoom(room)) {
+				throw new Error('error-invalid-room');
 			}
 
 			if (room.lastMessage?.token) {
-				return API.v1.failure('You cannot place chat on-hold, when the Contact has sent the last message');
+				throw new Error('error-contact-sent-last-message-so-cannot-place-on-hold');
 			}
 
 			if (room.onHold) {
-				return API.v1.failure('Room is already On-Hold');
+				throw new Error('error-room-is-already-onHold');
 			}
 
 			if (!room.open) {
-				return API.v1.failure('Room cannot be placed on hold after being closed');
+				throw new Error('This_conversation_is_already_closed');
 			}
 
-			const user = Meteor.user();
-			if (!user) {
-				return API.v1.failure('Invalid user');
+			if (!room.servedBy) {
+				throw new Error('error-unserved-rooms-cannot-be-placed-onhold');
 			}
 
-			const subscription = Subscriptions.findOneByRoomIdAndUserId(roomId, user._id, { _id: 1 });
+			const subscription = Subscriptions.findOneByRoomIdAndUserId(roomId, this.userId, { _id: 1 });
 			if (!subscription && !hasPermission(this.userId, 'on-hold-others-livechat-room')) {
-				return API.v1.failure('Not authorized');
+				throw new Error('Not_authorized');
 			}
 
-			const onHoldBy = { _id: user._id, username: user.username, name: (user as any).name };
+			const onHoldBy = { _id: this.userId, username: this.user.username, name: this.user.name };
 			const comment = TAPi18n.__('Omnichannel_On_Hold_manually', {
 				user: onHoldBy.name || `@${onHoldBy.username}`,
 			});
 
-			await LivechatEnterprise.placeRoomOnHold(room, comment, onHoldBy);
+			await OmnichannelEEService.placeRoomOnHold(room, comment, this.user);
+
+			return API.v1.success();
+		},
+	},
+);
+
+API.v1.addRoute(
+	'livechat/room.resumeOnHold',
+	{ authRequired: true, permissionsRequired: ['view-l-room'], validateParams: isLivechatRoomResumeOnHoldProps },
+	{
+		async post() {
+			const { roomId } = this.bodyParams;
+			if (!roomId || roomId.trim() === '') {
+				throw new Error('invalid-param');
+			}
+
+			const room = await LivechatRooms.findOneById<Pick<IOmnichannelRoom, '_id' | 't' | 'open' | 'onHold' | 'lastMessage' | 'servedBy'>>(
+				roomId,
+				{
+					projection: { _id: 1, t: 1, open: 1, onHold: 1, servedBy: 1 },
+				},
+			);
+			if (!room || !isOmnichannelRoom(room)) {
+				throw new Error('error-invalid-room');
+			}
+
+			if (!room.onHold) {
+				throw new Error('error-room-not-on-hold');
+			}
+
+			if (!room.open) {
+				throw new Error('This_conversation_is_already_closed');
+			}
+
+			const subscription = Subscriptions.findOneByRoomIdAndUserId(roomId, this.userId, { _id: 1 });
```

Comment:
and this

Reply:
https://github.com/RocketChat/Rocket.Chat/pull/28306/files#r1136560792